server/src/main/java/com/google/fhir/gateway/BearerAuthorizationInterceptor.java

```diff
@@ -192,12 +214,42 @@ public boolean authorizeRequest(RequestDetails requestDetails) {
       } else {
         reader = HttpUtil.readerFromEntity(entity);
       }
+
+      if (isEventLoggingEnabled) {
+        Reference agentUserWho = outcome.getUserWho(requestDetailsReader);
+        if (agentUserWho != null) {
+
+          StringWriter responseStringWriter = new StringWriter();
+          reader.transferTo(responseStringWriter);
+          String responseStringContent = responseStringWriter.toString();
+
+          Header contentLocationHeader = response.getFirstHeader(CONTENT_LOCATION_HEADER);
+
+          AuditEventHelper auditEventHelper =
+              new AuditEventHelperImpl(
+                  requestDetailsReader,
+                  responseStringContent,
+                  contentLocationHeader != null ? contentLocationHeader.getValue() : null,
+                  agentUserWho,
+                  authorizationDto.getDecodedJWT(),
```

Comment:
In some of the cases above, `decodedJWT` can be `null`? Do we properly handle this case in `AuditEventHelperImpl`? Please also add `@Nullable` for this parameter of the `AuditEventHelperImpl` constructor to document this (also we want to enable null-checker soon).

Also you can use a factory pattern to make it easier to mock this when writing tests for `BearerAuthorizationInterceptor`.

Reply:
I think the `null` handling part of this is done; not sure what are your thoughts on the factory pattern. If it is not needed for the unit-test that I asked for, we can skip that suggestion for now.